tests/new_unit_tests/Commands/Connection/SetCommandTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Pantheon\Terminus\UnitTests\Commands\Connection;
+
+use Pantheon\Terminus\Commands\Connection\SetCommand;
+use Prophecy\Prophet;
+
+/**
+ * Test suite for class for Pantheon\Terminus\Commands\Connection\SetCommand
+ */
+class SetCommandTest extends ConnectionCommandTest
+{
+    /**
+     * Test suite setup
+     *
+     * @return void
+     */
+    protected function setup()
+    {
+        parent::setUp();
+
+        $this->command = new SetCommand($this->getConfig());
+        $this->command->setLogger($this->logger);
+    }
+
+    /**
+     * Exercises connection:set git mode
+     *
+     * @return void
+     *
+     * @vcr site_set-connection-mode_git
+     */
+    public function testConnectionSetGit()
```

Comment:
As discussed in the team meeting this morning, I am going to wait until the DI approach is merged in #1249 and then I'll be able to differentiate the units from the behaviour tests to make this clearer/more useful.